File: testers/grpc/scala/generated-and-checked-in/com/example/grpc/ListOrdersRequest.scala
package com.example.grpc

import com.google.protobuf.CodedInputStream
import com.google.protobuf.CodedOutputStream
import com.google.protobuf.WireFormat
import io.grpc.MethodDescriptor.Marshaller
import java.io.ByteArrayInputStream
import java.io.IOException
import java.io.InputStream
import java.lang.RuntimeException

case class ListOrdersRequest(
  customerId: String,
  pageSize: Int
) {
  @throws[IOException]
  def writeTo(output: CodedOutputStream): Unit = {
    output.writeString(1, this.customerId)
    output.writeInt32(2, this.pageSize)
  }

  def getSerializedSize: Int = {
    var size: Int = 0
    size = size + CodedOutputStream.computeStringSize(1, this.customerId)
    size = size + CodedOutputStream.computeInt32Size(2, this.pageSize)
    return size
  }
}

object ListOrdersRequest {
  given marshaller: Marshaller[ListOrdersRequest] = {
    new Marshaller[ListOrdersRequest] {
      override def stream(value: ListOrdersRequest): InputStream = {
        val bytes = Array.ofDim[Byte](value.getSerializedSize)
        val cos = CodedOutputStream.newInstance(bytes)
        try {
          value.writeTo(cos)
          cos.flush()
        } catch {
          case e: IOException => throw new RuntimeException(e)
        } 
        return new ByteArrayInputStream(bytes)
      }
      override def parse(stream: InputStream): ListOrdersRequest = {
        try {
          return ListOrdersRequest.parseFrom(CodedInputStream.newInstance(stream))
        } catch {
          case e: IOException => throw new RuntimeException(e)
        } 
      }
    }
  }

  @throws[IOException]
  def parseFrom(input: CodedInputStream): ListOrdersRequest = {
    var customerId: String = ""
    var pageSize: Int = 0
    while (!input.isAtEnd()) {
      val tag = input.readTag()
      if (WireFormat.getTagFieldNumber(tag) == 1) { customerId = input.readString() }
      else if (WireFormat.getTagFieldNumber(tag) == 2) { pageSize = input.readInt32() }
      else { input.skipField(tag) }
    }
    return new ListOrdersRequest(customerId, pageSize)
  }
}

File: testers/grpc/java/generated-and-checked-in/com/example/grpc/ListOrdersRequest.java
package com.example.grpc;

import com.google.protobuf.CodedInputStream;
import com.google.protobuf.CodedOutputStream;
import com.google.protobuf.WireFormat;
import io.grpc.MethodDescriptor.Marshaller;
import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.io.InputStream;

public record ListOrdersRequest(String customerId, Integer pageSize) {
  public ListOrdersRequest withCustomerId(String customerId) {
    return new ListOrdersRequest(customerId, pageSize);
  }

  public ListOrdersRequest withPageSize(Integer pageSize) {
    return new ListOrdersRequest(customerId, pageSize);
  }

  public static Marshaller<ListOrdersRequest> MARSHALLER =
      new Marshaller<ListOrdersRequest>() {
        @Override
        public InputStream stream(ListOrdersRequest value) {
          var bytes = new byte[value.getSerializedSize()];
          var cos = CodedOutputStream.newInstance(bytes);
          try {
            value.writeTo(cos);
            cos.flush();
          } catch (IOException e) {
            throw new RuntimeException(e);
          }
          return new ByteArrayInputStream(bytes);
        }

        @Override
        public ListOrdersRequest parse(InputStream stream) {
          try {
            return ListOrdersRequest.parseFrom(CodedInputStream.newInstance(stream));
          } catch (IOException e) {
            throw new RuntimeException(e);
          }
        }
      };

  public static ListOrdersRequest parseFrom(CodedInputStream input) throws IOException {
    String customerId = "";
    Integer pageSize = 0;
    while (!input.isAtEnd()) {
      var tag = input.readTag();
      if (WireFormat.getTagFieldNumber(tag) == 1) {
        customerId = input.readString();
      } else if (WireFormat.getTagFieldNumber(tag) == 2) {
        pageSize = input.readInt32();
      } else {
        input.skipField(tag);
      }
      ;
    }
    ;
    return new ListOrdersRequest(customerId, pageSize);
  }

  public Integer getSerializedSize() {
    Integer size = 0;
    size = size + CodedOutputStream.computeStringSize(1, this.customerId());
    size = size + CodedOutputStream.computeInt32Size(2, this.pageSize());
    return size;
  }

  public void writeTo(CodedOutputStream output) throws IOException {
    output.writeString(1, this.customerId());
    output.writeInt32(2, this.pageSize());
  }
}
